pkg/operator/config.go

```diff
@@ -82,6 +98,25 @@ func getProviderControllerFromImages(platform configv1.PlatformType, images Imag
 	}
 }
 
+// When the platform type is BareMetal, MAO is also responsible for deploying
+// the baremetal-operator pod
+func getAdditionalProviderImages(platform configv1.PlatformType, images Images) (map[string]string, error) {
+        var baremetalImages map[string]string
+
+	switch platform {
+	case configv1.BareMetalPlatformType:
+                baremetalImages["BaremetalOperator"] = images.BaremetalOperator
```

Comment:
also, can we please move this to an struct as we do for controllers?

Reply:
Done.